core/src/data/literal.rs

```diff
@@ -164,11 +164,26 @@ impl<'a> Literal<'a> {
                 }
             }
             (Null, Number(_)) | (Number(_), Null) | (Null, Null) => Ok(Literal::Null),
-            _ => Err(LiteralError::UnsupportedBinaryArithmetic(
-                format!("{:?}", self),
-                format!("{:?}", other),
-            )
-            .into()),
+            _ => Err(Self::unsupported_binary_op(self, BinaryOperator::Divide, other)?.into()),
+        }
+    }
+
+    pub fn bitwise_and(&self, other: &Literal<'a>) -> Result<Literal<'static>> {
+        match (self, other) {
+            (Number(l), Number(r)) => match (l.to_i64(), r.to_i64()) {
+                (Some(l), Some(r)) => match (l & r).to_bigint() {
```

Comment:
Rather than `to_bigint()`, this looks that `BigDecimal::from` would work.

ref. https://docs.rs/bigdecimal/latest/bigdecimal/struct.BigDecimal.html#impl-From%3C%26i64%3E-for-BigDecimal